eng/pipelines/templates/steps/publish-artifact.yml

```diff
@@ -0,0 +1,27 @@
+# This step is used to prevent duplication of artifact publishes when there is an issue that would prevent the overall success of the job.
+# Ensuring that we only publish when successful (and two a differently named artifact otherwise) will allow easy retry on a build pipeline
```

Comment:
This is already in eng/common.

Reply:
Removed as per our discussion. It was incorrectly added.